### PR TITLE
Remove baseProtocol and baseDomain

### LINKED ISSUE #333: Explore whether baseProtocol config option can be removed
Ziggy only deals with Laravel named routes, which as far as I know can only be `http` and `https`. If that's the case, URL protocols probably don't need to be configurable.

See https://github.com/DefinitelyTyped/DefinitelyTyped/pull/48210#discussion_r495505368

### PR DESCRIPTION
**Pull request by bakerkretzmar**
This PR removes the `baseProtocol` and `baseDomain` properties on the Ziggy configuration object. `baseDomain` actually wasn't used anywhere, and `baseProtocol` was inferred from `baseUrl`, so now we just use that directly.

Closes #333.

### CODE REVIEW

File: src/Ziggy.php [hunk 1]
```diff
@@ -22,12 +20,7 @@ public function __construct(string $group = null, string $url = null)
         $this->group = $group;
 
         $this->baseUrl = rtrim($url ?? config('app.url', url('/')), '/');
-
-        tap(parse_url($this->baseUrl), function ($url) {
-            $this->baseProtocol = $url['scheme'] ?? 'http';
-            $this->baseDomain = $url['host'] ?? '';
-            $this->basePort = $url['port'] ?? null;
-        });
+        $this->basePort = parse_url($this->baseUrl)['port'] ?? null;
```
Comment (jakebathman) [inline src/Ziggy.php:23]: Just wanna make sure, same as the other PR, that this gets updated to `$this->url` in whatever order these PRs are merged. Not necessarily here, probably in #338, but don't wanna forget.

File: src/js/route.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ class Route {
         // If  we're building just a path there's no origin, otherwise: if this route has a
         // domain configured we construct the origin with that, if not we use the app URL
         const origin = !this.config.absolute ? '' : this.definition.domain
-            ? `${this.config.baseProtocol}://${this.definition.domain}${this.config.basePort ? `:${this.config.basePort}` : ''}`
+            ? `${this.config.baseUrl.match(/^\w+:\/\//)[0]}${this.definition.domain}${this.config.basePort ? `:${this.config.basePort}` : ''}`
```
Comment (jakebathman) [inline src/js/route.js:31]: Same here, `this.config.url`

**Review by jakebathman — CHANGES_REQUESTED** [review]
Looks good. I marked two places where `baseUrl` is new in this PR just to highlight the order of PR merging with #338. Otherwise good to go 👍

File: src/Ziggy.php (hunk 1 above)
Comment (bakerkretzmar) [reply src/Ziggy.php:23]: 👍🏻

**Pull request merged**